code/game/machinery/hologram.dm

```diff
@@ -289,33 +291,34 @@ Possible to do for anyone motivated enough:
 	This may change in the future but for now will suffice.*/
 	if(user.eyeobj.loc != src.loc)//Set client eye on the object if it's not already.
 		user.eyeobj.setLoc(get_turf(src))
-	else if(!masters[user])//If there is no hologram, possibly make one.
+	else if(!LAZYLEN(masters) || !masters[user])//If there is no hologram, possibly make one.
 		activate_holo(user)
 	else//If there is a hologram, remove it.
 		clear_holo(user)
 
 /obj/machinery/holopad/process()
-	for(var/I in masters)
-		var/mob/living/master = I
-		var/mob/living/silicon/ai/AI = master
-		if(!istype(AI))
-			AI = null
-
-		if(!QDELETED(master) && !master.incapacitated() && master.client && (!AI || AI.eyeobj))//If there is an AI attached, it's not incapacitated, it has a client, and the client eye is centered on the projector.
-			if(is_operational())//If the  machine has power.
-				if(AI)	//ais are range based
-					if(get_dist(AI.eyeobj, src) <= holo_range)
-						continue
-					else
-						var/obj/machinery/holopad/pad_close = get_closest_atom(/obj/machinery/holopad, holopads, AI.eyeobj)
-						if(get_dist(pad_close, AI.eyeobj) <= holo_range)
-							var/obj/effect/overlay/holo_pad_hologram/h = masters[master]
-							unset_holo(master)
-							pad_close.set_holo(master, h)
+	if(LAZYLEN(masters))

```

Comment:
If you let the masters definition stay as `var/list/masters = list()`, you can remove this if check and probably a few other places where LAZY wrangling happens. You can't avoid the byond built-in init either way because of the armor list. As a whole, it's probably good enough if it works

Reply:
but i spent so much time lazy-izing it